api/v1alpha3/cluster_types.go

```diff
@@ -57,6 +62,18 @@ type ClusterSpec struct {
 
 // ANCHOR_END: ClusterSpec
 
+// ANCHOR: PostApplyAddon
+
+// PostApplyAddon specifies the addon's Secret parameters.
+type PostApplyAddon struct {
+	// Name is the name of the secret.
+	Name string `json:"name,omitempty"`
+	// Namespace is the namespace of the secret.
+	Namespace string `json:"namespace,omitempty"`
+}
```

Comment:
As mentioned above, we can probably remove the annotation and store a timestamp within this struct